### PR TITLE
Filter `Content-Length` requests

### PR DESCRIPTION
**Pull request by legionth**
Based on #21 

Added stream to filter 'Content-Length' requests.

### CODE REVIEW

**Comment by legionth** [comment]
Ping @clue

File: src/LengthLimitedStream.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    /** @internal */
+    public function handleData($data)
+    {
+        if ($this->transferredLength === $this->maxLength) {
+            if ($this->maxLength === 0) {
+                $this->emit('end', array());
+            }
+            $this->stream->removeListener('data', array($this, 'handleData'));
```
Comment (clue) [inline src/LengthLimitedStream.php:74]: This means the data listener will only be removed once there's *another* useless data event?

File: src/HttpServer.php [hunk 2]
```diff
@@ -125,18 +125,18 @@ public function handleBody(RequestInterface $request, ConnectionInterface $conne
...
+        $bodyBuffer = '';
 
-        $connection->on('data', $listener);
+        $stream->on('data', function ($data) use (&$bodyBuffer) {
+            $bodyBuffer .= $data;
+        });
+
+        $stream->on('end', function () use (&$bodyBuffer, &$listener, $connection, $that, $request) {
+            $request = $request->withBody(RingCentral\Psr7\stream_for($bodyBuffer));
+            $that->handleRequest($connection, $request);
```
Comment (clue) [inline src/HttpServer.php:138]: Functionally LGTM :+1: However, the exact same code also appears above for the chunked decoder? (DRY)

**Review by clue — CHANGES_REQUESTED** [review]
Nice :+1:

**Comment by clue** [comment]
Also, looking at the slightly bigger picture here, what does this PR do (PR title appears unclear)? Can you add some new tests for this?

File: src/HttpServer.php (hunk 2 above)
Comment (legionth) [reply src/HttpServer.php:138]: Will be handled in the next request. This will obsolete anyway.

**Comment by legionth** [comment]
Ping @clue

File: src/HttpServer.php (hunk 2 above)
Comment (clue) [reply src/HttpServer.php:138]: *okay* :+1:

**Review by clue — APPROVED** [review]
LGTM except for the PR title :+1:

**Pull request merged**